pkg/user/reaper/group_test.go

```diff
@@ -0,0 +1,143 @@
+			objects: []runtime.Object{
+				&authorizationapi.ClusterRoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-no-subjects"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{},
+				},
+				&authorizationapi.ClusterRoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-one-subject"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup", Kind: "Group"}},
+				},
+				&authorizationapi.ClusterRoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-mismatched-subject"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup"}, {Name: "mygroup", Kind: "User"}, {Name: "mygroup", Kind: "Other"}},
+				},
+			},
+			expected: [][]string{
+				{"update", "", "clusterrolebindings", "binding-one-subject"},
+				{"delete", "", "groups", "mygroup"},
+			},
+		},
+		{
+			name:  "namespaced bindings",
+			group: "mygroup",
+			objects: []runtime.Object{
+				&authorizationapi.RoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-no-subjects", Namespace: "ns1"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{},
+				},
+				&authorizationapi.RoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-one-subject", Namespace: "ns2"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup", Kind: "Group"}},
+				},
+				&authorizationapi.RoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-mismatched-subject", Namespace: "ns3"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup"}, {Name: "mygroup", Kind: "User"}, {Name: "mygroup", Kind: "Other"}},
+				},
+			},
+			expected: [][]string{
+				{"update", "ns2", "rolebindings", "binding-one-subject"},
+				{"delete", "", "groups", "mygroup"},
+			},
+		},
+		{
+			name:  "sccs",
+			group: "mygroup",
+			objects: []runtime.Object{
+				&kapi.SecurityContextConstraints{
+					ObjectMeta: kapi.ObjectMeta{Name: "scc-no-subjects"},
+					Groups:     []string{},
+				},
+				&kapi.SecurityContextConstraints{
+					ObjectMeta: kapi.ObjectMeta{Name: "scc-one-subject"},
+					Groups:     []string{"mygroup"},
+				},
+				&kapi.SecurityContextConstraints{
+					ObjectMeta: kapi.ObjectMeta{Name: "scc-mismatched-subjects"},
+					Users:      []string{"mygroup"},
+					Groups:     []string{"mygroup2"},
+				},
+			},
+			expected: [][]string{
+				{"update", "", "securitycontextconstraints", "scc-one-subject"},
+				{"delete", "", "groups", "mygroup"},
+			},
+		},
+	}
+
+	accessor := meta.NewAccessor()
+
+	for _, test := range tests {
+		tc := testclient.NewSimpleFake(test.objects...)
+		ktc := ktestclient.NewSimpleFake(test.objects...)
+
+		actual := [][]string{}
+		reactor := func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
It seems like you really want to whitelist a few expected actions and cause unexpected actions to fail. why not build the reactors to do that?